app/src/main/java/com/duckduckgo/app/browser/RealWebViewCompatWrapper.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2025 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser
+
+import android.annotation.SuppressLint
+import androidx.webkit.ScriptHandler
+import androidx.webkit.WebViewCompat
+import com.duckduckgo.app.browser.api.WebViewCapabilityChecker
+import com.duckduckgo.app.browser.api.WebViewCapabilityChecker.WebViewCapability
+import com.duckduckgo.browser.api.webviewcompat.WebViewCompatWrapper
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+import kotlinx.coroutines.withContext
+import logcat.LogPriority.ERROR
+import logcat.asLog
+import logcat.logcat
+
+@SuppressLint(
+    "RequiresFeature",
+    "AddWebMessageListenerUsage",
+    "AddDocumentStartJavaScriptUsage",
+    "RemoveWebMessageListenerUsage",
+)
+@ContributesBinding(AppScope::class)
+class RealWebViewCompatWrapper @Inject constructor(
```

Comment:
Similar to another comment, is this something we want people to be able to use or force them to go through the plugin?